utils/src/main/java/org/corfudb/utils/lock/persistence/LockStore.java

```diff
@@ -118,6 +130,33 @@ public boolean acquire(LockId lockId) throws LockStoreException {
         }
     }
 
+    @VisibleForTesting
+    public void forceAcquire(LockId lockId) throws LockStoreException {
+        CorfuStoreMetadata.Timestamp timestamp = corfuStore.getTimestamp();
+        Optional<LockData> lockInDatastore = get(lockId, timestamp);
+        if (!lockInDatastore.isPresent()) {
+            LockData newLockData = LockData.newBuilder()
+                    .setLockId(lockId)
+                    .setLeaseOwnerId(clientId)
+                    .setLeaseRenewalNumber(0)
+                    .setLeaseAcquisitionNumber(0)
+                    .build();
+            // if no lock present acquire(create) the lock in datastore
+            create(lockId, newLockData, timestamp);
+            log.info("Lock: {} Client:{} acquired lock. No pre-existing lease in datastore.", lockId, clientId);
+        }
+        else {
+            LockData newLockData = LockData.newBuilder()
+                    .setLockId(lockId)
+                    .setLeaseOwnerId(clientId)
+                    .setLeaseRenewalNumber(0)
+                    .setLeaseAcquisitionNumber(lockInDatastore.get().getLeaseAcquisitionNumber() + 1)
+                    .build();
+            update(lockId, newLockData, timestamp);
+            log.info("Lock: {} Client:{} force-acquired lock.", lockId, clientId);
+        }
```

Comment:
what if the lock is acquired by another server and must be revoked?

Reply:
That's is why the locking timeouts are configured via LockConfig. And we delay all the timeouts so this does not happen.